nuxeo-ai-core/src/main/java/org/nuxeo/ai/services/PersistedConfigurationServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * (C) Copyright 2006-2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * Contributors:
+ *     anechaev
+ */
+package org.nuxeo.ai.services;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.ai.configuration.ThresholdConfiguratorDescriptor;
+import org.nuxeo.common.xmap.XMap;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.kv.KeyValueService;
+import org.nuxeo.runtime.kv.KeyValueStore;
+import org.nuxeo.runtime.model.DefaultComponent;
+import org.nuxeo.runtime.model.Descriptor;
+import org.nuxeo.runtime.model.impl.ExtensionDescriptorReader;
+
+public class PersistedConfigurationServiceImpl extends DefaultComponent implements PersistedConfigurationService {
+
+    private static final Logger log = LogManager.getLogger(PersistedConfigurationServiceImpl.class);
+
+    private static final ExtensionDescriptorReader reader = new ExtensionDescriptorReader();
+
+    public static final String KEY_VALUE_STORE = "aiConfigurationKVStore";
+
+    public static final String ALL_KEYS = "KEYS";
+
+    public PersistedConfigurationServiceImpl() {
+        this.register(ThresholdConfiguratorDescriptor.class);
```

Comment:
this part will have to be more generic - i let it this way for now (except if better idea ?)